### PR TITLE
feat: add github user extractor

### PR DESCRIPTION
**Pull request by ravisuhag**
This extractor pulls user data from github and creates a user metadata model.

### CODE REVIEW

File: core/extractor/extractor.go [hunk 1]
```diff
@@ -21,13 +21,12 @@ func (e *Extractor) Extract(name string, config map[string]interface{}) (data in
 	switch extr := extr.(type) {
 	case TableExtractor:
 		data, err = extr.Extract(config)
-		break
```
Comment (StewartJingga) [inline core/extractor/extractor.go:24]: TIL break is implicit in go.👌

**Comment by StewartJingga** [comment]
dont forget to list down your extractor [here](https://github.com/odpf/meteor/blob/main/docs/reference/extractors.md) as mentioned in [this guide](https://github.com/odpf/meteor/blob/main/docs/contribute/guide.md).

**Review by StewartJingga — CHANGES_REQUESTED** [review]
dont forget to list down your extractor [here](https://github.com/odpf/meteor/blob/main/docs/reference/extractors.md) as mentioned in [this guide](https://github.com/odpf/meteor/blob/main/docs/contribute/guide.md).

**Comment by ravisuhag** [comment]
> dont forget to list down your extractor [here](https://github.com/odpf/meteor/blob/main/docs/reference/extractors.md) as mentioned in [this guide](https://github.com/odpf/meteor/blob/main/docs/contribute/guide.md).

Done!!!
@StewartJingga 
Also, how about tests for this? Actually, Not much to test apart from mocking the Github client. Any suggestion?

**Comment by StewartJingga** [comment]
> > dont forget to list down your extractor [here](https://github.com/odpf/meteor/blob/main/docs/reference/extractors.md) as mentioned in [this guide](https://github.com/odpf/meteor/blob/main/docs/contribute/guide.md).
> 
> Done!!!
> @StewartJingga
> Also, how about tests for this? Actually, Not much to test apart from mocking the Github client. Any suggestion?

yeah been thinking about that as well, the best we could do is creating a mock and an interface of github client just to test if we are mapping the right data to meta.Table.

**Pull request merged**